src/main/java/entities/ComputerPlayer.java

```diff
@@ -0,0 +1,38 @@
+package entities;
+
+import java.util.*;
+
+public class ComputerPlayer extends Player {
+
+    String name;
+
+    /**
+     * Construct a ComputerPlayer with the given Hand.
+     * @param name The name to distinguish between the different ComputerPlayers.
+     * @param hand The Hand instance to be this ComputerPlayer's Hand.
+     */
+    public ComputerPlayer(String name, Hand hand) {
+        super(hand);
+        this.name = name;
+    }
+
+
+    /**
+     * Select a random valid Card to play in this Game
+     * @param game The Game instance this ComputerPlayer is a part of.
+     * @return a Card that is to be placed down in the Game.
+     * @throws NoValidCardException iff there is no valid Cards to play for the given Game.
+     */
+    public Card selectRandomCard(Game game) throws NoValidCardException {
```

Comment:
would returning a bool be better?

Reply:
Since we want a Card object to be returned, should we return null instead in the case there is no valid Card.